api/features/feature_segments/serializers.py

```diff
@@ -110,6 +112,13 @@ def validate(self, attrs):
                 "All feature segments must belong to the same feature & environment."
             )
 
+        environment = environments.pop()
+
+        if not self.context["request"].user.has_environment_permission(
```

Comment:
I don't really like that the permission check is in serializer instead of `permissions.py` but due to the structure of the request this feels a bit clean, what do you think?

Reply:
Hmm... yeah, interesting I guess this is because we don't have any context on which objects are being accessed before this point. 

I think you're right, that keeping it here makes the most sense.